zipline/testing/fixtures.py

```diff
@@ -315,72 +333,218 @@ def init_class_fixtures(cls):
+      :func:`zipline.pipeline.loaders.synthetic.make_minute_bar_data`.
+    - `cls.bcolz_minute_bar_reader` is a minute bar reader pointing to the
+      directory that was just written to.
 
+    See Also
+    --------
+    WithBcolzDailyBarReader
+    WithDataPortal
+    """
+    BCOLZ_MINUTE_BAR_PATH = 'minute_equity_pricing.bcolz'
+    BCOLZ_MINUTE_BAR_LOOKBACK_DAYS = 0
+    BCOLZ_MINUTE_BAR_USE_FULL_CALENDAR = False
+
+    @classmethod
+    def make_minute_bar_data(cls):
+        return create_minute_bar_data(
+            cls.env.minutes_for_days_in_range(
+                cls.bcolz_minute_bar_days[0],
+                cls.bcolz_minute_bar_days[-1],
+            ),
+            cls.asset_finder.sids,
+        )
+
+    @classmethod
+    def init_class_fixtures(cls):
+        super(WithBcolzMinuteBarReader, cls).init_class_fixtures()
+        cls.bcolz_minute_bar_path = p = cls.tmpdir.makedir(
+            cls.BCOLZ_MINUTE_BAR_PATH,
+        )
+        cls.bcolz_minute_bar_days = days = (
+            cls.env.trading_days
+            if cls.BCOLZ_MINUTE_BAR_USE_FULL_CALENDAR else
+            cls.env.days_in_range(
+                cls.env.trading_days[
+                    cls.env.get_index(cls.sim_params.period_start) -
+                    cls.BCOLZ_MINUTE_BAR_LOOKBACK_DAYS
+                ],
+                cls.sim_params.period_end,
+            )
+        )
         writer = BcolzMinuteBarWriter(
-            cls.env.first_trading_day,
-            cls.tempdir.path,
-            cls.env.open_and_closes.market_open,
-            cls.env.open_and_closes.market_close,
-            cls.BCOLZ_MINUTES_PER_DAY,
+            days[0],
+            p,
+            cls.env.open_and_closes.market_open.loc[days],
+            cls.env.open_and_closes.market_close.loc[days],
+            US_EQUITIES_MINUTES_PER_DAY
+        )
+        cls.bcolz_minute_bar_data = df_dict = cls.make_minute_bar_data()
+        for sid, df in iteritems(df_dict):
+            writer.write(sid, df)
+
+        cls.bcolz_minute_bar_reader = BcolzMinuteBarReader(p)
+
+
+class WithAdjustmentReader(WithBcolzDailyBarReader):
+    """
+    ZiplineTestCase mixin providing cls.adjustment_reader as a class level
+    fixture.
+
+    After init_class_fixtures has been called, `cls.adjustment_reader` will be
+    populated with a new SQLiteAdjustmentReader object. The data that will be
+    written can be passed by overriding `make_{field}_data` where field may
+    be `splits`, `mergers` `dividends`, or `stock_dividends`.
+    The daily bar reader used for this adjustment reader may be customized
+    by overriding `make_adjustment_writer_daily_bar_reader`. This is useful
+    to providing a `MockDailyBarReader`.
+
+    For more advanced configuration, `make_adjustment_writer` may be
+    overwritten directly.
+
+    See Also
+    --------
+    zipline.testing.MockDailyBarReader
+    """
+    @classmethod
+    def _make_data(cls):
```

Comment:
Thoughts on making this provide empty frames with the right dtypes instead of `None` as the value for "no data"?